packages/agreement-process/test/testAgreementConsumerDocuments.ts

```diff
@@ -1,97 +1,265 @@
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreementId,
+            agreement1.consumerDocuments[0].id,
+            authData
+          )
+        ).rejects.toThrowError(agreementNotFound(agreementId));
+      });
+
+      it("should throw an operationNotAllowed error when the requester is not the consumer or producer", async () => {
+        const authData = getRandomAuthData(generateId<TenantId>());
+
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreement1.id,
+            agreement1.consumerDocuments[0].id,
+            authData
+          )
+        ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+      });
+
+      it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
+        const authData = getRandomAuthData(agreement1.consumerId);
+        const agreementDocumentId = generateId<AgreementDocumentId>();
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreement1.id,
+            agreementDocumentId,
+            authData
+          )
+        ).rejects.toThrowError(
+          agreementDocumentNotFound(agreementDocumentId, agreement1.id)
+        );
+      });
     });
 
-    it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
-      const authData = getRandomAuthData(agreement1.consumerId);
-      const agreementDocumentId = generateId<AgreementDocumentId>();
-      await expect(
-        agreementService.getAgreementConsumerDocument(
-          agreement1.id,
-          agreementDocumentId,
-          authData
-        )
-      ).rejects.toThrowError(
-        agreementDocumentNotFound(agreementDocumentId, agreement1.id)
-      );
+    describe("add", () => {
+      it("should succeed on happy path", async () => {
+        const organizationId = generateId<TenantId>();
+        const authData: AuthData = {
+          ...getRandomAuthData(organizationId),
+          userRoles: ["admin"],
+        };
+        const agreement = getMockAgreement(
+          generateId<EServiceId>(),
+          organizationId
+        );
+
+        const consumerDocument = getMockConsumerDocument(agreement.id);
+        const correlationId = "9999";
+
+        await addOneAgreement(agreement, postgresDB, agreements);
+
+        await agreementService.addConsumerDocument(
+          agreement.id,
+          consumerDocument,
+          authData,
+          correlationId
+        );
+        const { data: payload } = await readLastAgreementEvent(
+          agreement.id,
+          postgresDB
+        );
+
+        const actualConsumerDocument = decodeProtobufPayload({
+          messageType: AgreementConsumerDocumentAddedV1,
+          payload,
+        });
+
+        expect(actualConsumerDocument).toMatchObject({
+          agreementId: agreement.id,
+          document: {
```

Comment:
Here you are comparing the `AgreementDocumentV1` (the protobuf generated model for agreement documents written in event payloads) with an `AgreementDocument`, the domain model we use in the app business logic. To make this more explicit in other tests we create the "expected" object as a domain model object, **then** in the `expect` we convert to protobuf to match the one actually in the payload:

Something like:

```ts
const expectedConsumerDocument: AgreementDocument = {
   ...consumerDocument,
   createdAt:  new Date(Number(actualConsumerDocument.document!.createdAt)),
 }  

expect(actualConsumerDocument).toMatchObject({
          agreementId: agreement.id,
          document: toAgreementDocumentV1(expectedConsumerDocument),
});
```